src/Pangraph.hs

```diff
@@ -135,41 +133,37 @@ vertexAttributes = vertexAttributes'
 edgeEndpoints :: Edge -> (Vertex, Vertex)
 edgeEndpoints = endpoints'
 
-{-| Returns the EdgeID if it has one. `Edge`s are given a new `VetexID` when they are passed and retrived from a `Pangraph`-}
+{-| Returns the EdgeID if it has one. 'Edge's are given a new 'EdgeID' when they are passed and retrived from a 'Pangraph'-}
 edgeID :: Edge -> Maybe EdgeID
 edgeID = edgeID'
 
--- | Returns a `VetexID`
+-- | Returns a 'VetexID'
 vertexID :: Vertex -> VertexID
 vertexID = vertexID'
 
 -- * Operations on Edge and Vertex
 
--- | Lookup Attributes by `Key`s
--- | Complexity: /O(n)/
+-- | Lookup Attributes by 'Key's Complexity: /O(n)/
 lookupVertexValues :: Key -> Vertex -> Maybe Value
 lookupVertexValues k v = lookup k (vertexAttributes v)
 
--- | Lookup Attributes by `Key`s
--- | Complexity: /O(n)/
+-- | Lookup Attributes by 'Key's Complexity: /O(n)/
 lookupEdgeValues :: Key -> Edge -> Maybe Value
 lookupEdgeValues k e = lookup k (edgeAttributes e)
 
 -- * Utility Operations
--- | Similar to `vertices` but returns an association list instead
+-- | Similar to 'vertices' but returns an association list instead
 vertexAssocList :: Pangraph -> [(VertexID, Vertex)]
 vertexAssocList p = Map.toList $ vertices' p
 
--- | Simlar to `edges` but returns an association list instead
+-- | Simlar to 'edges' but returns an association list instead
 edgeAssocList :: Pangraph -> [(EdgeID, Edge)]
 edgeAssocList p = Map.toList $ edges' p
 
--- | Returns a Bool representing whether a `Key` is present.
--- | Complexity /O(n)/
+-- | Returns a Bool representing whether a 'Key' is present. Complexity /O(n)/
```

Comment:
If you give the user `edgeAttributes` they would be able to implement this themselves, potentially faster than O(n). I suggest to drop this function, simplifying the API.